### PR TITLE
Teacher ip location

### PR DESCRIPTION
**Pull request by RyanNovas**
- adds ip_location table that is filled out whenever a teacher without an ip_location signs in
- includes a rake task to add an ip_location for all locationless teachers that are not affiliated with a school

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -304,6 +307,10 @@ def requires_password?
     permanent? && new_record?
   end
 
+  def create_ip_location
```
Comment (ddmck) [inline app/models/user.rb]: this needs to be removed

File: app/models/user.rb [hunk 2]
```diff
@@ -31,7 +30,11 @@ class User < ActiveRecord::Base
 
   has_and_belongs_to_many :schools
   has_and_belongs_to_many :districts
+  has_and_belongs_to_many :ip_locations
```
Comment (ddmck) [inline app/models/user.rb]: this needs to be removed as it now has_one :ip_location

File: app/workers/ip_location_worker.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class IpLocationWorker
+  include Sidekiq::Worker
+
+  def perform(id, ip_address)
+    @user = User.find(id)
+    location = Pointpin.locate(ip_address)
+    unless location["postcode"] == ("10005" || "11237" || "11385") #locations where we've had offices and therefore triggered a ton of false ip addresses
```
Comment (ddmck) [inline app/workers/ip_location_worker.rb]: Do we really need to set this on every sign in? We dont need to black list IPs if if is only set on creating the user

File: db/migrate/20160127210202_create_ip_locations_users.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class CreateIpLocationsUsers < ActiveRecord::Migration
```
Comment (ddmck) [inline db/migrate/20160127210202_create_ip_locations_users.rb]: this file can be deleted as ip_location now has user_id

File: lib/quill_authentication.rb [hunk 5]
```diff
@@ -26,6 +26,7 @@ def current_user
 
   def sign_in user
     remote_ip = (request.present? ? request.remote_ip : nil)
+    IpLocationWorker.perform_async(user.id, remote_ip) if (user.ip_location == nil) && (user.role != 'student')
```
Comment (ddmck) [inline lib/quill_authentication.rb]: This should be in sign up.

File: app/workers/ip_location_worker.rb (hunk 3 above)
Comment (RyanNovas) [inline app/workers/ip_location_worker.rb]: Not necessary on every sign in. I was thinking more about the rake task, which will have a ton of users with those ip_addresses, but i can filter them at that point.

**Comment by RyanNovas** [comment]
@ddmck fixed as per your comments (with the exception of removing the join table migration) -- i just ran a following one that dropped it as I'd already pushed.

**Comment by RyanNovas** [comment]
ready for review @ddmck

**Pull request merged**